app/lib/authenticated_system.rb

```diff
@@ -1,4 +1,8 @@
 module AuthenticatedSystem
+  def current_ability
+    @current_ability ||= Ability.new(current_user.present? ? current_user.decorate.object : nil)
```

Comment:
https://github.com/3scale/porta/pull/2108#discussion_r471570977

Reply:
i would go with this idea: https://github.com/3scale/porta/pull/2108#discussion_r471581718

this `current_user.decorate.object` does not look very smart.

Reply:
`current_user.decorate` will decorate the model again if it's not a `Decorator` instance yet, right? Wdyt about:

```ruby
Ability.new(current_user.try(:model) || nil)
```

Reply:
it is `current_user.try(:object)`, but imo, it should be then: `Ability.new(current_user.try(:object) || current_user)`

Reply:
@guicassolato @macejmic changed 😎

Reply:
From https://github.com/drapergem/draper/blob/v3.0.1/README.md

> When writing decorator methods you'll usually need to access the wrapped model. While you may choose to use delegation (covered below) for convenience, you can always use the object (or its alias model)

😉

Reply:
@guicassolato I'm sorry but I do not understand how that quote helps us in this case.

Reply:
just to be clear, I've done `Ability.new(current_user.try(:object) || current_user)` because if it does not respond to `object`, that means that `current_user` is either the User model or `nil`, and both should do in our case.